Common/Securities/Futures/FutureFilterUniverse.cs

```diff
@@ -0,0 +1,205 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+using System;
+using System.Collections.Generic;
+using QuantConnect.Data;
+using System.Linq;
+using System.Collections;
+using QuantConnect.Util;
+
+namespace QuantConnect.Securities
+{
+    /// <summary>
+    /// Represents futures symbols universe used in filtering.
+    /// </summary> 
+    public class FutureFilterUniverse : IDerivativeSecurityFilterUniverse
+    {
+        internal IEnumerable<Symbol> _allSymbols;
```

Comment:
Does this need to be exposed as `internal`? Looks like it provides a `GetEnumerator` method that exposes it already. Same goes for the other `internal` member variables in here. It doesn't look like anyone outside of this class references them, so they could easily be made private (and probably should be for data protection).